votor-messages/src/bls_message.rs

```diff
@@ -0,0 +1,183 @@
+//! Put BLS message here so all clients can agree on the format
+use {
+    crate::vote::Vote,
+    bitvec::prelude::*,
+    serde::{Deserialize, Serialize},
+    solana_bls_signatures::Signature as BLSSignature,
+    solana_sdk::{clock::Slot, hash::Hash},
+};
+
+/// The seed used to derive the BLS keypair
+pub const BLS_KEYPAIR_DERIVE_SEED: &[u8; 9] = b"alpenglow";
+
+/// Block, a (slot, hash) tuple
+pub type Block = (Slot, Hash);
+
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+/// BLS vote message, we need rank to look up pubkey
+pub struct VoteMessage {
+    /// The vote
+    pub vote: Vote,
+    /// The signature
+    pub signature: BLSSignature,
+    /// The rank of the validator
+    pub rank: u16,
+}
+
+/// Certificate details
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Deserialize, Serialize)]
+pub enum CertificateId {
```

Comment:
Should this be called `CertificateId`? This is really just the `Certificate` right?

Reply:
This is really Certificate now, if you call it CertificateId then I would expect the content somewhere.